### DISCUSSION TITLE
Import KeePass 2 XML with attachments

### DISCUSSION THREAD

**Discussion by tig987** [dani-garcia/vaultwarden#1693] [Q&A]
Hi,

I imported a KeePass 2 XML into the Vaultwarden and then recognized that attachments, that were present in Keepass entries, were not imported into the Vaultwarden.

Is this an expected behaviour or a bug?

**Comment by BlackDex** [comment]
This is a limitation of the bitwarden web-vault it doesn't support attachments in that way unfortunately.

**Reply by tig987** [reply]
But I am able to add attachments through the Web vault interface.

**Reply by BlackDex** [reply]
Yes you are, but that is a different feature.
See: https://github.com/bitwarden/web/issues/183 and https://github.com/bitwarden/web/issues/426

Currently Bitwarden doesn't support importing those items via the web-vault.
However, i see that someone did create a small tool to make that work: https://github.com/jampe/kp2bw

**Reply by ChillDuder** [reply]
i used that kp2bw and can recommend. worked fine.

maybe it's worth a note in wiki/importing data.. page.
